### PR TITLE
Make the remote name more unique

### PR DESCRIPTION
**Pull request by sstok**
|Q            |A   |
|---          |--- |
|Fixed Tickets|#438|
|License      |MIT |

Include the repository name and host(name) to ensure a proper uniqueness for the remote name.

Todo:

- [x] Update BranchDeleteCommand
- [x] Update BranchForkCommand
- [x] Update BranchMergeCommand
- [x] Update BranchPushCommand
- [x] Update BranchRemoteAddCommand
- [x] Update IssueTakeCommand
- [x] Update PullRequestCheckoutCommand
- [x] Update PullRequestFixerCommand
- [x] Update PullRequestMergeCommand
- [x] Update PullRequestSemVerCommand
- [x] Update PullRequestSquashCommand
- [x] Update PullRequestSwitchBaseCommand
- [x] Update FabbotIoCommand

### CODE REVIEW

**Comment by sstok** [comment]
@cordoval ping

**Comment by cordoval** [comment]
I will review, allow for some time

File: src/RemoteName.php [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+        if ($host) {
+            $host = trim(preg_replace('{^https?://}i', '', $host), '/');
+        }
+
+        $this->org = $org;
+        $this->name = $name;
+        $this->host = $host;
+
+        if (null !== $host) {
+            $hostString = (
```
Comment (cordoval) [inline src/RemoteName.php]: he he

File: tests/BaseTestCase.php [hunk 2]
```diff
@@ -166,6 +166,8 @@ protected function getGitConfigHelper()
     {
         $helper = $this->prophesize('Gush\Helper\GitConfigHelper');
         $helper->setHelperSet(Argument::any())->willReturn();
+        $helper->ensureRemoteExists(Argument::cetera())->will(function ($args) { return implode('_', $args); });
```
Comment (cordoval) [inline tests/BaseTestCase.php]: i am not sure i like prophecy anymore, but still pondering

File: tests/RemoteNameTest.php [hunk 3]
```diff
@@ -0,0 +1,34 @@
+<?php
```
Comment (cordoval) [inline tests/RemoteNameTest.php]: forgot the header

**Comment by cordoval** [comment]
i give a light review, i will review more tomorrow and the day after

File: tests/BaseTestCase.php (hunk 2 above)
Comment (sstok) [inline tests/BaseTestCase.php]: It's the only way we can mock things as PHPUnit mocks are only expectations and there is no way to say: return `...` when the parameters are `...`.

File: tests/RemoteNameTest.php (hunk 3 above)
Comment (sstok) [inline tests/RemoteNameTest.php]: fixed

**Comment by cordoval** [comment]
I think this PR introduces changes that are improvements that should be taken in, except for the part that it adds the name based on hosting and such. Let's split this into 2 PRs one with the improvements and oen just with the extended name. The former PR should be good. As per the second PR can be further discussed.

**Comment by sstok** [comment]
Basically the host appending is needed for the "enterprise" support, having a secondary repo in the same organization is not completely uncommon (Symfony uses it for security patches), but I agree you don't necessarily use Gush for this :) I'm happy with only including the host part in the remote name.

Looking at the current state, only `GitConfigHelper::ensureRemoteExists()` uses the RemoteName ValueObject, so it should be able to move it to that method 👍 

*Including the adapter-name is overkill, when you host multiple implementations on the same host-name you properly have a bigger problem.*🙅

**Comment by sstok** [comment]
Ah wait I now see what you mean 😃 only change the remote-name but keep the display as it was (organization only).

**Comment by sstok** [comment]
I changed the pull-request to only change the remote name but leave the display as it was.

**Comment by cordoval** [comment]
let's think this through some more, i am thinking on the use case where you add a remote that is from another clone, the same organization is the master origin, but someone else clone that and then you are recloning. Something smells here badly, i mean i am feeling like the complication from cordoval to cordoval_gush is going to break sooner rather than later. i also always picture cordoval as the perspective that gush initially took, meaning there is an origin (not upstream) and forks. The forks bear the name of the collaborators usernames. That is the rule. I feel like if we break that rule ...

**Comment by sstok** [comment]
Snap I forgot we only need the org and hostname (when not null), good catch 👍

**Comment by cordoval** [comment]
yeah it is not going to cut it, because it is too adhoc. What i wonder though is if the ensureRemoteExists warnings were already introduced into a separate PR, that we can do if needed.

**Comment by sstok** [comment]
If you have only one host that you are using, and not working with cross repositories (read-only to monolith) then the remote name as it is now is good.

Only need to warn when the remote already exists but doesn't meet the expectation.
Will do this in a new pr 👍  thanks for the review and suggestions.

**Pull request closed without merging**